File: GestaoLivrosApi/GestaoLivrosApi/Models/Books/EditBookDTO.cs
using System;
using FluentValidation;

namespace GestaoLivrosApi.Models.Books
{
	public class EditBookDTO
	{
        public int Id { get; set; }

        public long Isbn { get; set; }

        public string Name { get; set; } = null!;

        public int AuthorId { get; set; }

        public double Price { get; set; }

        public string? Image { get; set; }


        public class EditBookDTOValidator : AbstractValidator<EditBookDTO>
        {
            public EditBookDTOValidator()
            {
                RuleFor(b => b.Isbn).NotNull().WithMessage("Insira o ISBN do livro").GreaterThan(0).WithMessage("Insira um ISBN superior a 0");
                RuleFor(b => b.Price).NotNull().WithMessage("Insira o Preço do livro").GreaterThan(0).WithMessage("Insira um valor superior a 0");
                RuleFor(b => b.Name).NotNull().WithMessage("Insira o Nome do livro");
                RuleFor(b => b.AuthorId).NotNull().WithMessage("Insira o Autor do livro");


            }
        }
	}
}

File: frontend/src/models/Books/EditBookDTO.ts
import Joi from "joi";

export class EditBookDTO{
    id: number = 0;
    isbn: number = 0;
    name: string = "";
    authorId: number = 0;
    authorName: string ="";
    price: number = 0;
    image: string | null = null;
}

export const EditBookDTOSchema = Joi.object({
    isbn: Joi.number().min(0).required().messages({"number.base": "Isbn do livro deve ser preenchido", "number.min":"Isbn deve ser superior a 0"}),
    name: Joi.string().messages({"string.base":"Nome do livro deve ser preenchido"}),
    authorId: Joi.number().min(0).messages({"number.base":"Autor de ser selecionado"}),
    price: Joi.number().min(0).messages({"number.base":"Preço deve ser preenchido", "number.min":"Preço deve ser superior a 0"}),
    
})